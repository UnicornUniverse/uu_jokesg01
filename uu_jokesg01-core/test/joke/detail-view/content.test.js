import UU5 from "uu5g04";
import { shallow, mount, wait } from "uu5g05-test";
import Content from "../../../src/joke/detail-view/content.js";

const jokeDataObject = {
  state: "ready",
  data: {
    id: "123",
    name: "Test joke",
    visibility: false,
    text: "Best joke ever",
    imageUrl: "https://via.placeholder.com/300x300",
    image: "1234",
    ratingCount: 5,
    averageRating: 2.5,
    categoryIdList: ["1", "2"],
    sys: {
      cts: "2020-01-01T00:00:00.000Z",
    },
  },
};

const jokesPermission = {
  joke: {
    canAddRating: () => true,
    canManage: () => true,
  },
};

const jokesDataObject = {
  state: "ready",
  data: {
    categoryList: [
      { id: "1", name: "Category 1" },
      { id: "2", name: "Category 2" },
      { id: "3", name: "Category 3" },
    ],
  },
};

const parentStyle = {
  paddingLeft: 8,
  paddingRight: 8,
  paddingTop: 8,
  paddingBottom: 8,
};

describe(`UuJokesCore.Joke.DetailView.Content`, () => {
  it(`default props`, () => {
    const wrapper = shallow(
      <Content
        jokesDataObject={jokesDataObject}
        jokeDataObject={jokeDataObject}
        jokesPermission={jokesPermission}
        parentStyle={parentStyle}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it(`onAddRating callback`, async () => {
    const handleAddRating = jest.fn(() => null);

    const jokesPermission = {
      joke: {
        canAddRating: () => false,
        canManage: () => true,
      },
    };

    const wrapper = mount(
      <Content
        jokesDataObject={jokesDataObject}
        jokeDataObject={jokeDataObject}
        jokesPermission={jokesPermission}
        parentStyle={parentStyle}
        onAddRating={handleAddRating}
      />
    );

    const ratingElement = wrapper.find(UU5.Bricks.Icon).first();
    ratingElement.simulate("click");

    await wait();
    expect(handleAddRating).toHaveBeenCalledTimes(0);
  });

  it(`no add rating permission`, async () => {
    const handleAddRating = jest.fn(() => null);

    const wrapper = mount(
      <Content
        jokesDataObject={jokesDataObject}
        jokeDataObject={jokeDataObject}
        jokesPermission={jokesPermission}
        parentStyle={parentStyle}
        onAddRating={handleAddRating}
      />
    );

    const ratingElement = wrapper.find(UU5.Bricks.Icon).first();
    ratingElement.simulate("click");

    await wait();
    expect(handleAddRating).toHaveBeenCalledTimes(1);
  });
});
