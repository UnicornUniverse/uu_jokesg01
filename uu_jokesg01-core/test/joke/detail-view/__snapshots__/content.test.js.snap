// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UuJokesCore.Joke.DetailView.Content default props 1`] = `
<div
  className="uujokescore-joke-detailview-1cqsrz9"
>
  <Uu5Elements.withTooltip(Uu5Elements.Text)
    category="interface"
    className="uujokescore-joke-detailview-d0ziyl"
    colorScheme="building"
    segment="content"
    type="medium"
  >
    Best joke ever
  </Uu5Elements.withTooltip(Uu5Elements.Text)>
  <img
    alt="Test joke"
    className="uujokescore-joke-detailview-fu2utz"
    src="https://via.placeholder.com/300x300"
  />
  <Uu5Elements.Line
    significance="subdued"
  />
  <InfoLine
    className="uujokescore-joke-detailview-10yg275"
  >
    Category 1, Category 2
  </InfoLine>
  <InfoLine
    className="uujokescore-joke-detailview-10yg275"
  >
    <Uu5Elements.DateTime
      dateFormat="short"
      timeFormat="none"
      value="2020-01-01T00:00:00.000Z"
    />
  </InfoLine>
  <InfoLine
    className="uujokescore-joke-detailview-10yg275"
  >
    5 votes
  </InfoLine>
  <Uu5Elements.Box
    className="uujokescore-joke-detailview-xa1zn1"
    significance="distinct"
  >
    <span>
      <Plus4U5Elements.PersonItem />
    </span>
    <Rating
      onClick={[Function]}
      value={2.5}
    />
  </Uu5Elements.Box>
</div>
`;
